sources/packages/api/src/esdc-integration/disbursement-receipt-integration/disbursement-receipt-processing.service.ts

```diff
@@ -0,0 +1,188 @@
+import { Injectable } from "@nestjs/common";
+import {
+  DisbursementReceipt,
+  DisbursementReceiptValue,
+  DisbursementSchedule,
+  User,
+} from "../../database/entities";
+import { InjectLogger } from "../../common";
+import { LoggerService } from "../../logger/logger.service";
+import {
+  ConfigService,
+  DisbursementScheduleService,
+  DisbursementReceiptService,
+} from "../../services";
+import { ESDCIntegrationConfig } from "../../types";
+import { DisbursementReceiptIntegrationService } from "./disbursement-receipt-integration.service";
+import { ProcessSFTPResponseResult } from "../models/esdc-integration.model";
+import { DisbursementReceiptDownloadResponse } from "./models/disbursement-receipt-integration.model";
+
+/**
+ * Manages the process to import disbursement receipt files which are
+ * downloaded from SFTP location.
+ */
+@Injectable()
+export class DisbursementReceiptProcessingService {
+  private readonly esdcConfig: ESDCIntegrationConfig;
+  constructor(
+    config: ConfigService,
+    private readonly integrationService: DisbursementReceiptIntegrationService,
+    private readonly disbursementScheduleService: DisbursementScheduleService,
+    private readonly disbursementReceiptService: DisbursementReceiptService,
+  ) {
+    this.esdcConfig = config.getConfig().ESDCIntegration;
+  }
+
+  /**
+   * Process all the available disbursement receipt files in SFTP location.
+   * Once the file is processed, it gets deleted.
+   * @param auditUserId
+   * @returns Summary details of the processing.
+   */
+  async process(auditUserId: number): Promise<ProcessSFTPResponseResult> {
+    const result = new ProcessSFTPResponseResult();
+
+    // Get the list of all files from SFTP ordered by file name.
+    const fileSearch = new RegExp(
+      `^${this.esdcConfig.environmentCode}EDU.PBC.DIS.D[\w]*\.[0-9]*`,
```

Comment:
can you please confirm whether the filename that we read from the server has a file extension line .001 or 2 or .300 etc?. If there is a specific type of filename like .txt or .dat, please append them at the end.